src/mappings/DAOTracker/mapping.ts

```diff
@@ -0,0 +1,146 @@
+import { Address, ipfs, store } from '@graphprotocol/graph-ts';
+import {
+  BlacklistDAO,
+  DAOTracker,
+  ResetDAO,
+  TrackDAO,
+} from '../../types/DAOTracker/DAOTracker';
+import {
+  AvatarContract,
+  BlacklistedDAO,
+  ContractInfo,
+  DAOTrackerContract,
+  ResetDAO as ResetDAOEntity,
+  UControllerOrganization,
+} from '../../types/schema';
+import { createTemplate, equalStrings, fetchTemplateName } from '../../utils';
+
+export function getDAOTrackerContract(address: Address): DAOTrackerContract {
+  let daoTracker = DAOTrackerContract.load(address.toHex()) as DAOTrackerContract;
+  if (daoTracker == null) {
+    daoTracker = new DAOTrackerContract(address.toHex());
+    daoTracker.address = address;
+    let daoTrackerSC = DAOTracker.bind(address);
+    daoTracker.owner = daoTrackerSC.owner();
+    daoTracker.save();
+  }
+  return daoTracker;
+}
+
+export function handleTrackDAO(event: TrackDAO): void {
+  // Ensure the DAOTrackerContract has been added to the store
+  getDAOTrackerContract(event.address);
+
+  let avatar = event.params._avatar;
+  let controller = event.params._controller;
+  let reputation = event.params._reputation;
+  let daoToken = event.params._daoToken;
+  let sender = event.params._sender;
+  let arcVersion = event.params._arcVersion;
+
+  // If the avatar already exists, early out
+  if (AvatarContract.load(avatar.toHex()) != null) {
+    return;
+  }
+
+  // If the sender of the 'track' call is the DaoCreator contract, use its arcVersion
+  let daoCreatorInfo = ContractInfo.load(sender.toHex());
+  if (daoCreatorInfo != null) {
+    if (equalStrings(daoCreatorInfo.name, 'DaoCreator')) {
+      arcVersion = daoCreatorInfo.version;
+    }
+  }
+
+  let avatarTemplate = fetchTemplateName('Avatar', arcVersion);
+  let controllerTemplate = fetchTemplateName('Controller', arcVersion);
+  let reputationTemplate = fetchTemplateName('Reputation', arcVersion);
+  let daoTokenTemplate = fetchTemplateName('DAOToken', arcVersion);
+
+  let missingTemplate = avatarTemplate == null ||
+                        reputationTemplate == null ||
+                        daoTokenTemplate == null;
+
+  let universalController = UControllerOrganization.load(controller.toHex()) != null;
```

Comment:
`!= null` ?

Reply:
If loading the UControllerOrganization != null, it means we're using the universalController. If we're not using the universal controller, we want to (1) check to see if we're missing the template for the controller version, and (2) create the template if it exists.

Reply:
oh, the syntax is exceptional .
`let universalController = (UControllerOrganization.load(controller.toHex()) != null);` this might be more readable ..